sdk/src/main/java/com/horizen/transaction/SidechainCoreTransactionSerializer.java

```diff
@@ -1,12 +1,40 @@
 package com.horizen.transaction;
 
+import com.horizen.box.Box;
+import com.horizen.box.data.*;
+import com.horizen.proof.Proof;
+import com.horizen.proof.ProofSerializer;
+import com.horizen.proof.Signature25519Serializer;
+import com.horizen.proposition.Proposition;
+import com.horizen.utils.DynamicTypedSerializer;
+import com.horizen.utils.ListSerializer;
 import scorex.util.serialization.Reader;
 import scorex.util.serialization.Writer;
 
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+
+import static com.horizen.transaction.BoxTransaction.MAX_TRANSACTION_UNLOCKERS;
+import static com.horizen.transaction.SidechainCoreTransaction.SIDECHAIN_CORE_TRANSACTION_VERSION;
+
 public final class SidechainCoreTransactionSerializer implements TransactionSerializer<SidechainCoreTransaction>
 {
     private static SidechainCoreTransactionSerializer serializer;
 
+    // Serializers definition
+    private final static ListSerializer<BoxData<Proposition, Box<Proposition>>> boxesDataSerializer = new ListSerializer<>(
+            new DynamicTypedSerializer<>(new HashMap<Byte, BoxDataSerializer>() {{
+                put((byte)1, ZenBoxDataSerializer.getSerializer());
+                put((byte)2, WithdrawalRequestBoxDataSerializer.getSerializer());
+                put((byte)3, ForgerBoxDataSerializer.getSerializer());
+            }}, new HashMap<>()), MAX_TRANSACTION_UNLOCKERS);
```

Comment:
It was a mistake before. For output data, that represents new boxes, we need to use `MAX_TRANSACTION_NEW_BOXES`